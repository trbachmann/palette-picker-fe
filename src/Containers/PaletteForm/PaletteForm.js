import React, { Component } from 'react';
import { connect } from 'react-redux';
import { postPalette } from '../../Thunks/postPalette';
import { fetchProjectPalettes } from '../../Thunks/fetchProjectPalettes';

export class PaletteForm extends Component{
  constructor() {
    super();
    this.state = {
      name: '',
      projectId: ''
    };
  };

  createPalette = () => {
    const { name } = this.state;
    const {currentPalette} = this.props
    let paletteToAdd = currentPalette.reduce((acc, color, index) => {
      acc[`color${index+1}`] = color.hex
      return acc
    }, {})
    paletteToAdd = {...paletteToAdd, name}
    return paletteToAdd
  }

  handleChange = (event) => {
    const { value } = event.target;
    this.setState({ name: value });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const palette = this.createPalette();
    const  projectId  = parseInt(this.state.projectId)
    this.props.fetchProjectPalettes(projectId);
    this.props.postPalette(projectId, palette);
    this.setState({name: '', projectId: ''})
  };

  makeOptions = () => {
    const projectOptions = this.props.projects.map(project => {
      return(
        <option value={project.id} key={project.id}>{project.name}</option>
      )
    })
    return projectOptions
  }

  setProject = (event) => {
    const { value } = event.target;
    this.setState({projectId: value})
  }

  render() {
    return(
      <form 
        onSubmit={this.handleSubmit} 
        className='PaletteForm--form'>
        <select onChange={this.setProject} value={this.state.projectId}>
          <option value='' defaultValue>Choose a Project</option>
          {this.makeOptions()}
        </select>
        <input 
          placeholder='Enter New Palette Name'
          name='name'
          value={this.state.name}
          onChange={this.handleChange}/>
        <button>Save Palette</button>
      </form>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  postPalette: (projectId, palette) => dispatch(postPalette(projectId, palette)),
  fetchProjectPalettes: (projectId) => dispatch(fetchProjectPalettes(projectId))
});

export const mapStateToProps = state => ({
  currentPalette: state.currentPalette,
  projects: state.projects
})

export default connect(mapStateToProps, mapDispatchToProps)(PaletteForm);