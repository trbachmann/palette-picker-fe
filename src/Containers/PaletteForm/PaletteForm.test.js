import React from 'react';
import { shallow } from 'enzyme';
import { PaletteForm } from './PaletteForm';

describe('PaletteForm', () => {
  describe('PaletteForm component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<PaletteForm/>);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});