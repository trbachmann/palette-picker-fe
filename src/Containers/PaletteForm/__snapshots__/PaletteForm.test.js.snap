// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteForm PaletteForm component should match the snapshot 1`] = `
<form
  className="PaletteForm--form"
  onSubmit={[Function]}
>
  <select
    onChange={[Function]}
    value=""
  >
    <option
      defaultValue={true}
      value=""
    >
      Choose a Project
    </option>
    <option
      key="1"
      value={1}
    >
      Game Time
    </option>
    <option
      key="2"
      value={2}
    >
      Movie Tracker
    </option>
    <option
      key="3"
      value={3}
    >
      Trapper Keeper
    </option>
  </select>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Enter New Palette Name"
    value=""
  />
  <button>
    Save Palette
  </button>
</form>
`;
